pkg/kudoctl/util/kudo/kudo.go

```diff
@@ -253,81 +244,57 @@ func (c *Client) UpdateInstance(instanceName, namespace string, operatorVersion
 	if err != nil {
 		return err
 	}
-	_, err = c.kudoClientset.KudoV1beta1().Instances(namespace).Patch(context.TODO(), instanceName, types.MergePatchType, serializedPatch, v1.PatchOptions{})
+	newInstance, err := c.kudoClientset.KudoV1beta1().Instances(namespace).Patch(context.TODO(), instanceName, types.MergePatchType, serializedPatch, v1.PatchOptions{})
 	if err != nil {
 		return err
 	}
 	if !wait {
 		return nil
 	}
-	return c.WaitForInstance(instanceName, namespace, oldInstance, waitTime)
+	return c.WaitForInstance(newInstance, waitTime)
 }
 
 // WaitForInstance waits for instance to be "complete".
 // It uses controller-runtime `wait.PollImmediate`, the function passed to it returns done==false if it isn't done.
-// For a situation where there is no previous state (like install), the "lastPlanStatus" will be nil until the manager
-// sets it, then it's state will be watched (see InInstanceDone for more detail)
-// For a situation where there is previous state (like update, upgrade, plan trigger) than it is important AND required
-// that the "oldInstance" be provided.  Without it, it is possible for this function to be "racy" and "flaky" meaning the
-// "current" status could be the old "done" status or the new status... it's hard to know.  If the oldInstance is provided
-// the wait will then initially wait for the "new" plan to activate then return when completed.
+// The passed updated instance has the scheduled plan with its UID saved in Spec.PlanExecution field (if a new plan
+// was scheduled as the result of the update). This method will wait for this plan to become terminal.
 // The error is either an error in working with kubernetes or a wait.ErrWaitTimeout
-func (c *Client) WaitForInstance(name, namespace string, oldInstance *v1beta1.Instance, timeout time.Duration) error {
+func (c *Client) WaitForInstance(instance *v1beta1.Instance, timeout time.Duration) error {
 	// polling interval 1 sec
+	plan := instance.Spec.PlanExecution.PlanName
+
 	interval := 1 * time.Second
 	return wait.PollImmediate(interval, timeout, func() (done bool, err error) {
-		instance, err := c.GetInstance(name, namespace)
+		i, err := c.GetInstance(instance.Name, instance.Namespace)
 		if err != nil {
 			return false, err
 		}
 
-		return c.IsInstanceDone(instance, oldInstance)
+		return c.IsInstanceDone(i, plan)
 	})
 }
 
 // IsInstanceDone provides a check on instance to see if it is "finished" without retries
 // oldInstance is nil if there is no previous instance
```

Comment:
This comment should probably be adjusted. I have no idea what "finished" means in the context here... Also, oldInstance isn't a parameter anymore